components/MoveTimeModal.vue

```diff
@@ -0,0 +1,158 @@
+  secondsToLivesplitFormat,
+  stringTimeToSeconds
+}                                        from '~/util/durations';
+import {splitFileIsModified, selectTime, Segment, OptionalRealAndGameTime} from '~/util/splits';
+import {Component, Prop, mixins}         from 'nuxt-property-decorator';
+import BaseModal                         from '~/components/BaseModal.vue';
+import Multiselect                       from 'vue-multiselect';
+import {whithLoad}                       from '~/util/loading';
+import store                             from '~/util/store';
+
+@Component({components: {Multiselect}})
+export default class MoveTimeModal extends mixins(BaseModal) {
+  modalRef: string = 'MoveTimeModal';
+
+  transferTime: number = 0;
+
+  transferSplitName: string = "";
+
+  @Prop()
+  splits!: Segment[];
+
+  @Prop()
+  currentSplitIndex!: number;
+
+  get segmentGold() {
+    const gold = selectTime(this.currentSplit.BestSegmentTime);
+    return stringTimeToSeconds(gold || "0.0.0.0");
+  }
+
+  get splitOptions() {
+    if (this.currentSplitIndex == 0)
+      return [this.nextSplit.Name];
+
+    const lastSplitIndex = this.splits.length - 1;
+    if (this.currentSplitIndex == lastSplitIndex)
+      return [this.previousSplit.Name];
+
+    return [this.previousSplit.Name, this.nextSplit.Name];
+  }
+
+  get timeValid() {
+    if (this.transferTime == 0 || this.transferTime > this.segmentGold || this.transferSplitName == '') {
+      return false;
+    }
+
+    return true;
+  }
+
+  get invalidTimeTooltip() {
+    if (this.transferTime == 0)
+      return 'You must add a time.';
+
+    if (this.transferSplitName == '')
+      return 'You need select a split to transfer to.';
+
+    if (this.transferTime > this.segmentGold)
+      return `You cannot transfer more than ${secondsToFormattedString(this.segmentGold)}.`
+
+    return "";
+  }
+
+  get currentSplit() {
+    return this.splits[this.currentSplitIndex];
+  }
+
+  get nextSplit() {
+    return this.splits[this.currentSplitIndex + 1];
+  }
+
+  get previousSplit() {
+    return this.splits[this.currentSplitIndex - 1];
+  }
+
+  get chosenSplitFunction() {
+    if (this.transferSplitName === this.nextSplit?.Name)
+      return () => this.moveTimeNext();
+    
+    if (this.transferSplitName === this.previousSplit?.Name)
+      return () => this.moveTimePrevious();
+  }
+
+  moveTransferTime(chosenTime: OptionalRealAndGameTime, isCurrentSplit: boolean) {
+    let transferTime = this.transferTime;
+    if (isCurrentSplit)
+      transferTime = -transferTime;
+
+    const newRealTime = stringTimeToSeconds(chosenTime.RealTime || "0.0.0.0") + transferTime;
+    chosenTime.RealTime = secondsToLivesplitFormat(newRealTime);
+
+    if (store.state.hasGameTime) {
+      const newGameTime = stringTimeToSeconds(chosenTime?.GameTime || "0.0.0.0") + transferTime;
+      chosenTime.GameTime = secondsToLivesplitFormat(newGameTime);
+    }
+  }
+
+  moveTimePrevious() {
+    whithLoad(() => {
```

Comment:
If we get a bit more general and call these two variables 
```TS
const removeTimeSplit = this.currentSplit.SegmentHistory?.Time;
const addTimeSplit = this.previousSplit.SegmentHistory?.Time;
```
then `moveTimePrevious` and `moveTimeNext` are basically the exact same functions with these two variable swapped right? So if we make the variables parameters, we can use just one function and do 
```TS
moveTimePrevious() {
    // moveTime is my new general function
    this.moveTime(this.currentSplit, this.previousSplit);
}

moveTimeNext() {
    this.moveTime(this.previousSplit, this.currentSplit);
}
```
... and then we probably won't need these two functions any more actually, we'll just be able to call this new `moveTime` function from wherever we need. This will also make `moveTime` much easier to move to a web worker because it won't have any more references to `this` since the data is passed as explicit parameters

Reply:
Wait no I'm dumb, it's even easier than this, what you would do is 
```TS
moveTimePrevious() {
    // moveTime is my new general function
    this.moveTime(this.currentSplit, this.previousSplit);
}

moveTimeNext() {
    this.moveTime(this.currentSplit, this.nextSplit);
}
```
it's not even a matter of swapping the arguments, just change the split that's no the current one. Which means that the parameters of `moveTime` could just be `currentTimeSplit` and `otherTimeSplit` or something (instead of `removeTimeSplit` and `addTimeSplit` that I suggested earlier, but these names still work too)